client/state/data-layer/wpcom/read/site/post-email-subscriptions/index.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * External Dependencies
+ */
+import { omit } from 'lodash';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal Dependencies
+ */
+import {
+	READER_SUBSCRIBE_TO_NEW_POST_EMAIL,
+	READER_UNSUBSCRIBE_TO_NEW_POST_EMAIL,
+	READER_UPDATE_NEW_POST_EMAIL_SUBSCRIPTION,
+} from 'state/action-types';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { subscribeToNewPostEmails, unsubscribeToNewPostEmails } from 'state/reader/follows/actions';
+import { errorNotice } from 'state/notices/actions';
+
+function requestPostEmailSubscription( { dispatch }, action, next ) {
+	dispatch( http( {
+		method: 'POST',
+		path: `/read/site/${ action.payload.blogId }/post_email_subscriptions/new`,
+		onSuccess: action,
+		onFailure: action,
+	} ) );
+	next( action );
+}
+
+function receivePostEmailSubscription( store, action, next, response ) {
+	// validate that it worked
+	// if it did, just swallow this response, as we don't need to pass it along.
+	console.log( response );
+}
+
+function receivePostEmailSubscriptionError( { dispatch }, action, next, error ) {
+	dispatch( errorNotice( translate( 'Sorry, we had a problem subscribing. Please try again.' ) ) );
+	// dispatch an unsubscribe
+	next( unsubscribeToNewPostEmails( action.payload.blogId ) );
```

Comment:
this piece of the data layer is listening for it, to kick off a request to remove the subscription from the server. Here we only want to update the reducers, but not kick off the http request. We can use next to bypass the data layer and only pass the action on to the reducers.